### ISSUE TITLE
[8.1.0-SNAPSHOT] Fleet Server can't enroll: FAILED: Missed two check-ins

### ISSUE THREAD

**Issue by mtojek** [elastic/fleet-server#1129]
Hi,

we adopted `elastic-package` to use predefined [agent policies](https://github.com/elastic/elastic-package/blob/main/internal/profile/_static/kibana_config_8x.yml#L23) and confirmed with @juliaElastic that we're ready for switch (main branch is green).

Since yesterday we're facing problems with enrollment:

```
Attaching to elastic-package-stack_fleet-server_1
[36mfleet-server_1               |[0m Performing setup of Fleet in Kibana
[36mfleet-server_1               |[0m 
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:21.515Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":587},"message":"Spawning Elastic Agent daemon as a subprocess to complete bootstrap process.","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:21.712Z","log.origin":{"file.name":"application/application.go","file.line":78},"message":"Detecting execution mode","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:21.713Z","log.origin":{"file.name":"application/application.go","file.line":98},"message":"Agent is in Fleet Server bootstrap mode","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:22.031Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":62},"message":"Starting stats endpoint","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:22.031Z","log.origin":{"file.name":"application/fleet_server_bootstrap.go","file.line":134},"message":"Agent is starting","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:22.031Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":64},"message":"Metrics endpoint listening on: /usr/share/elastic-agent/state/data/tmp/elastic-agent.sock (configured: unix:///usr/share/elastic-agent/state/data/tmp/elastic-agent.sock)","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:22.034Z","log.origin":{"file.name":"application/fleet_server_bootstrap.go","file.line":144},"message":"Agent is stopped","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:22.141Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":48},"message":"New State ID is Wb5PhdQX","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:22.142Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":49},"message":"Converging state requires execution of 1 step(s)","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:22.733Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-02-03T08:05:22Z - message: Application: fleet-server--8.1.0-SNAPSHOT[]: State changed to STARTING: Starting - type: 'STATE' - sub_type: 'STARTING'","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:22.735Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":66},"message":"Updating internal state","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:05:24.523Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":792},"message":"Fleet Server - Starting","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"warn","@timestamp":"2022-02-03T08:06:27.040Z","log.origin":{"file.name":"status/reporter.go","file.line":236},"message":"Elastic Agent status changed to: 'degraded'","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:06:27.040Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-02-03T08:06:27Z - message: Application: fleet-server--8.1.0-SNAPSHOT[]: State changed to DEGRADED: Missed last check-in - type: 'STATE' - sub_type: 'RUNNING'","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:07:21.517Z","log.origin":{"file.name":"cmd/run.go","file.line":203},"message":"Shutting down Elastic Agent and sending last events...","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:07:21.519Z","log.origin":{"file.name":"operation/operator.go","file.line":223},"message":"waiting for installer of pipeline 'default' to finish","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:07:21.520Z","log.origin":{"file.name":"process/app.go","file.line":176},"message":"Signaling application to stop because of shutdown: fleet-server--8.1.0-SNAPSHOT","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"error","@timestamp":"2022-02-03T08:07:27.047Z","log.origin":{"file.name":"status/reporter.go","file.line":236},"message":"Elastic Agent status changed to: 'error'","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"error","@timestamp":"2022-02-03T08:07:27.047Z","log.origin":{"file.name":"log/reporter.go","file.line":36},"message":"2022-02-03T08:07:27Z - message: Application: fleet-server--8.1.0-SNAPSHOT[]: State changed to FAILED: Missed two check-ins - type: 'ERROR' - sub_type: 'FAILED'","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:07:51.570Z","log.origin":{"file.name":"status/reporter.go","file.line":236},"message":"Elastic Agent status changed to: 'online'","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:07:51.570Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-02-03T08:07:51Z - message: Application: fleet-server--8.1.0-SNAPSHOT[]: State changed to STOPPED: Stopped - type: 'STATE' - sub_type: 'STOPPED'","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:07:51.570Z","log.origin":{"file.name":"cmd/run.go","file.line":211},"message":"Shutting down completed.","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m {"log.level":"info","@timestamp":"2022-02-03T08:07:51.570Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":66},"message":"Stats endpoint (/usr/share/elastic-agent/state/data/tmp/elastic-agent.sock) finished: accept unix /usr/share/elastic-agent/state/data/tmp/elastic-agent.sock: use of closed network connection","ecs.version":"1.6.0"}
[36mfleet-server_1               |[0m Error: fleet-server failed: context canceled
[36mfleet-server_1               |[0m For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.1/fleet-troubleshooting.html
[36mfleet-server_1               |[0m Error: enrollment failed: exit status 1
[36mfleet-server_1               |[0m For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.1/fleet-troubleshooting.html
```

More logs: https://beats-ci.elastic.co/job/Ingest-manager/job/integrations/job/main/98/artifact/build/elastic-stack-dump/synthetics/logs/

It affects the Integrations main branch, incl. synthetics, containerd, etc.

Steps to reproduce:

```
elastic-package stack update -v -d --version 8.1.0-SNAPSHOT
elastic-package stack up -v -d --version 8.1.0-SNAPSHOT
```

Thanks for any help with investigating this problem. 

cc @jlind23 @joshdover

**Comment by joshdover** [comment]
Curious that there's no logs from Fleet server related to the policy it selected.

One possible workaround could be to add `FLEET_SERVER_POLICY_ID=fleet-server-managed-ep` to the env vars for Fleet Server here: https://github.com/elastic/elastic-package/blob/main/internal/profile/_static/docker-compose-stack.yml#L79

But we should figure out what the root issue here is regardless of if the workaround works.

**Comment by mtojek** [comment]
Hey @joshdover, we tried it with Julia while working on migration to hosted policies. With `FLEET_SERVER_POLICY_ID=fleet-server-managed-ep` it will break compatibility with 7.x stack. If it isn't really necessary, I would postpone introducing this env. Otherwise, we'll have to implement some hack/forked stack.

**Comment by mtojek** [comment]
We did some investigation and it seems that the root cause is in the Elastic Agent Docker image. The last stable one we managed to build is this one: https://github.com/elastic/elastic-package/pull/683 (`@sha256:a5c580573376d65ed2eba92d359b411cdae4bf52745af8e3bb8c0c91f8ce53a5`).

It maps into:

```
elastic-agent@cd978fd268e2:~$ ./elastic-agent diagnostics
elastic-agent  version: 8.1.0
               build_commit: 56b227d00945ae97d7e8663df048c29f311b8894  build_time: 2022-02-01 07:01:00 +0000 UTC  snapshot_build: true
Applications:
  *  name: metricbeat  route_key: default
     error: Get "http://unix/": dial unix /usr/share/elastic-agent/state/data/tmp/default/metricbeat/metricbeat.sock: connect: no such file or directory
```

**Comment by mtojek** [comment]
We suspect that problem might have been introduced with this PR: https://github.com/elastic/beats/pull/29031

cc @ph @blakerouse

**Comment by ph** [comment]
I will take a look, but I think the PR you have is the only major thing that I know that could have impact the agent.

**Comment by juliaElastic** [comment]
this change might be related as well: https://github.com/elastic/kibana/issues/108252

when starting elastic-package locally, I see this in fleet-server logs:
`Kibana Fleet setup failed: http POST request to http://kibana:5601/api/fleet/setup fails: Forbidden: <nil>. Response: {"statusCode":403,"error":"Forbidden","message":"Forbidden"}`

what is more, .fleet-agents index is not created, I have a suspicion that fleet-server might not have access to fleet API at all

**Comment by joshdover** [comment]
I don't think it's related to that change. I've been able to successfully run this on `main` with any issues as a manual test:

```
# Create new elastic/fleet-server token
curl --request POST \
  --url http://localhost:9200/_security/service/elastic/fleet-server/credential/token \
  -u elastic:changeme

# Copy token response into authz header below
curl --request POST \
  --url http://localhost:5601/api/fleet/setup \
  --header 'authorization: Bearer <token>' \
  --header 'content-type: application/json' \
  --header 'kbn-xsrf: x' 
```

Do we need to update the token that we're using? Maybe our manual hardcoded token isn't working anymore due to a change in ES?

**Comment by mtojek** [comment]
@joshdover In [this PR](https://github.com/elastic/elastic-package/pull/683) I forced the specific Docker image for Elastic Agent and it passed. Elasticsearch, Kibana images were the same.

**Comment by ph** [comment]
OK, I went through the all commits in fleet-server, the latest commit that add actual code in the server is 4 days ago, https://github.com/elastic/fleet-server/pulls?q=is%3Apr+is%3Amerged I am going to concentrate on the Agent side of things.

**Comment by ph** [comment]
This was a really deep rabbit hole, I took some time to have a running testing and a fast environment to debug using the `AGENT_DROP_PATH` and having part of Elastic-Agent precompiled and just building the current platform and architecture of the Elastic Agent. Looking at the behavior of the `elastic-package` the Fleet server was really waiting for an initial configuration. Looking at the Fleet in Kibana, I was able to see that the system was stuck in _waiting_ the first enrollment. The other behavior of Fleet-Server was the heavy CPU usage, maybe the fleet-server is stuck in a live loop? I also tested outside of the Docker environment, and the bug was present there two.

Because of this, I've initially thought the problem was into fleet-server. I've bisected the last good commit of the fleet server and the bug was still present. Now, everything was showing a problem with the Elastic Agent side, so I also did a bisect of the last good build to the working version. I was able to narrow it down to the actual APM instrumentation.

Looking at the implementations the traces should have been disabled by default and not impact any behavior of the agent. If I remove the whole PR the Elastic Agent was able to do the initial enrollment into Fleet without any problems. I look more closely at the code I've tried to remove the code of the gRPC interceptor but it did not fix the situation. I've decided to revert the whole implementation of the APM instrumentation and we will need to look into it more. I've notice that importing `apmgprc` has an init side effect. I've removed the import but it didn't fix the problem.


Reverting the PR was noted a simple revert,  another pull request applied after had a conflicting change.

Looking at that PR, it was green except for the e2e CI, if the latter was working I am confident it would have caught that issues.



**Action items:**

- Write a better documentation for testing this scenario.
- Make the build way quicker, I takes a lot of minutes to have a working binary.
- Write a post-morterm
- E2E testing needs to be enabled back or have a simple job with the `elastic-package` to bring the stack.
- Investigate high CPU usage
- Investigate APM Instrumentation of Elastic Agent with @stuartnelson3 
-

**Comment by mtojek** [comment]
Let's keep it open until we confirm that it fixed.

**Comment by ph** [comment]
interesting that the issue was closed from a forked repository? I don't remember seeing that ever.

**Comment by axw** [comment]
I'm seeing what appears to be the same issue with `docker.elastic.co/beats/elastic-agent:8.2.0-5d69c4c3-SNAPSHOT`, which is built from https://github.com/elastic/beats/commit/5529c31cf1bd68bf2ad089ef747186f9510ff3f1 (after the revert).

To reproduce, clone https://github.com/elastic/apm-server/pull/7227 and run `docker-compose up -d`. Fleet Server fails to enroll, 

Logs:

```
$ docker-compose logs fleet-server
Attaching to apm-server_fleet-server_1
fleet-server_1      | Requesting service_token from Kibana.
fleet-server_1      | Created service_token named: token-1644310962231
fleet-server_1      | Performing setup of Fleet in Kibana
fleet-server_1      | 
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:43.549Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":572},"message":"Spawning Elastic Agent daemon as a subprocess to complete bootstrap process.","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:43.766Z","log.origin":{"file.name":"application/application.go","file.line":68},"message":"Detecting execution mode","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:43.774Z","log.origin":{"file.name":"application/application.go","file.line":88},"message":"Agent is in Fleet Server bootstrap mode","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:44.557Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":744},"message":"Waiting for Elastic Agent to start Fleet Server","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:44.576Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":62},"message":"Starting stats endpoint","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:44.576Z","log.origin":{"file.name":"application/fleet_server_bootstrap.go","file.line":131},"message":"Agent is starting","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:44.576Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":64},"message":"Metrics endpoint listening on: /usr/share/elastic-agent/state/data/tmp/elastic-agent.sock (configured: unix:///usr/share/elastic-agent/state/data/tmp/elastic-agent.sock)","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:44.577Z","log.origin":{"file.name":"application/fleet_server_bootstrap.go","file.line":141},"message":"Agent is stopped","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:46.731Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":48},"message":"New State ID is nLbqrZoq","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:46.731Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":49},"message":"Converging state requires execution of 1 step(s)","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:48.286Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-02-08T09:02:48Z - message: Application: fleet-server--8.2.0-SNAPSHOT[]: State changed to STARTING: Starting - type: 'STATE' - sub_type: 'STARTING'","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:48.287Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":66},"message":"Updating internal state","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:49.368Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-02-08T09:02:49Z - message: Application: fleet-server--8.2.0-SNAPSHOT[]: State changed to STARTING: Waiting on default policy with Fleet Server integration - type: 'STATE' - sub_type: 'STARTING'","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:02:50.566Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":777},"message":"Fleet Server - Waiting on default policy with Fleet Server integration","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:04:43.551Z","log.origin":{"file.name":"cmd/run.go","file.line":185},"message":"Shutting down Elastic Agent and sending last events...","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:04:43.551Z","log.origin":{"file.name":"operation/operator.go","file.line":216},"message":"waiting for installer of pipeline 'default' to finish","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:04:43.551Z","log.origin":{"file.name":"process/app.go","file.line":176},"message":"Signaling application to stop because of shutdown: fleet-server--8.2.0-SNAPSHOT","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:04:45.053Z","log.origin":{"file.name":"cmd/run.go","file.line":193},"message":"Shutting down completed.","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:04:45.053Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-02-08T09:04:45Z - message: Application: fleet-server--8.2.0-SNAPSHOT[]: State changed to STOPPED: Stopped - type: 'STATE' - sub_type: 'STOPPED'","ecs.version":"1.6.0"}
fleet-server_1      | {"log.level":"info","@timestamp":"2022-02-08T09:04:45.053Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":66},"message":"Stats endpoint (/usr/share/elastic-agent/state/data/tmp/elastic-agent.sock) finished: accept unix /usr/share/elastic-agent/state/data/tmp/elastic-agent.sock: use of closed network connection","ecs.version":"1.6.0"}
fleet-server_1      | Error: fleet-server failed: context canceled
fleet-server_1      | For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.2/fleet-troubleshooting.html
fleet-server_1      | Error: enrollment failed: exit status 1
fleet-server_1      | For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.2/fleet-troubleshooting.html
```

**Comment by mtojek** [comment]
@ph Could you please check what is the status of the elastic-agent Docker image? the issue still persists in [Integrations](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations/detail/main/125/pipeline).

**Comment by ph** [comment]
@mtojek I am taking another look.

**Comment by ph** [comment]
@mtojek Looking at the [failure of the ci](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations/detail/main/125/pipeline), this concern the 8.1 snapshots, and well I didn't merge https://github.com/elastic/beats/pull/30209, I will double-check the failures and merge it. Is there a job that test on master?

**Comment by axw** [comment]
@ph if you don't care about running the specific steps that @mtojek mentioned: the steps I listed in https://github.com/elastic/fleet-server/issues/1129#issuecomment-1032421904 are for main (8.2.0-SNAPSHOT).

**Comment by ph** [comment]
```
unix /usr/share/elastic-agent/state/data/tmp/elastic-agent.sock: use of closed network connection","ecs.version":"1.6.0"}
fleet-server_1 | Error: fleet-server failed: context canceled
fleet-server_1 | For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.2/fleet-troubleshooting.html
fleet-server_1 | Error: enrollment failed: exit status 1
fleet-server_1 | For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.2/fleet-troubleshooting.html
fleet-server_1 | {"log.level":"info","@timestamp":"2022-02-08T16:42:09.461Z","log.origin":
```

**Comment by ph** [comment]
Build is `5529c31cf1bd68bf2ad089ef747186f9510ff3f1` and does include the revert

```
❯ git show 5529c31cf1bd68bf2ad089ef747186f9510ff3f1                                                                                                                                                                             [11:48:31]
commit 5529c31cf1bd68bf2ad089ef747186f9510ff3f1 (HEAD)
Author: Elastic Machine <elasticmachine@users.noreply.github.com>
Date:   Mon Feb 7 10:17:17 2022 -0600

    [Release] update version to next minor 8.2.0 (#30160)

diff --git a/libbeat/version/version.go b/libbeat/version/version.go
index 873ae40db0..38249106a4 100644
--- a/libbeat/version/version.go
+++ b/libbeat/version/version.go
@@ -18,4 +18,4 @@
 // Code generated by dev-tools/set_version
 package version
 
-const defaultBeatVersion = "8.1.0"
+const defaultBeatVersion = "8.2.0
```

This is indeed a strange behavior, because I was able to reproduce the bug everytime with the instrumention commit and not without it.

**Comment by mtojek** [comment]
I retriggered the main job. Let's see what's the current status: [link](https://beats-ci.elastic.co/job/Ingest-manager/job/integrations/job/main/126/)

**Comment by ph** [comment]
Interesting logs in the Kibana side, not sure why we have multiple `fleet-setup completed statement`

```
[2022-02-08T16:39:23.863+00:00][INFO ][status] Kibana is now degraded (was available)

[2022-02-08T16:39:24.670+00:00][INFO ][plugins.fleet] Beginning fleet setup

[2022-02-08T16:39:24.759+00:00][INFO ][plugins.fleet] Fleet setup completed

[2022-02-08T16:39:24.798+00:00][INFO ][plugins.fleet] Beginning fleet setup

[2022-02-08T16:39:24.890+00:00][INFO ][plugins.fleet] Fleet setup completed

[2022-02-08T16:39:29.403+00:00][INFO ][status] Kibana is now available (was degraded)

[2022-02-08T17:00:25.540+00:00][INFO ][status] Kibana is now degraded (was available)

[2022-02-08T17:00:29.298+00:00][INFO ][status] Kibana is now available (was degraded)

[2022-02-08T17:02:38.529+00:00][INFO ][plugins.fleet] Beginning fleet setup

[2022-02-08T17:02:38.615+00:00][INFO ][plugins.fleet] Fleet setup completed

[2022-02-08T17:02:38.632+00:00][INFO ][plugins.fleet] Beginning fleet setup

[2022-02-08T17:02:38.718+00:00][INFO ][plugins.fleet] Fleet setup completed

[2022-02-08T17:07:14.094+00:00][INFO ][plugins.fleet] Beginning fleet setup

[2022-02-08T17:07:14.145+00:00][INFO ][plugins.fleet] Fleet setup completed

[2022-02-08T17:07:14.168+00:00][INFO ][plugins.fleet] Beginning fleet setup

[2022-02-08T17:07:14.217+00:00][INFO ][plugins.fleet] Fleet setup completed

[2022-02-08T17:14:45.142+00:00][INFO ][plugins.fleet] Beginning fleet setup

[2022-02-08T17:14:45.208+00:00][INFO ][plugins.fleet] Fleet setup completed

[2022-02-08T17:14:45.226+00:00][INFO ][plugins.fleet] Beginning fleet setup

[2022-02-08T17:14:45.349+00:00][INFO ][plugins.fleet] Fleet setup completed

[2022-02-08T17:25:17.553+00:00][ERROR][plugins.taskManager] Failed to poll for work: Error: work has timed out

[2022-02-08T17:25:17.580+00:00][INFO ][status] Kibana is now degraded (was available)

[2022-02-08T17:25:18.802+00:00][WARN ][plugins.kibanaUsageCollection] Average event loop delay threshold exceeded 350ms. Received 10813.265237333333ms. See https://ela.st/kibana-scaling-considerations for more information about scaling Kibana.

[2022-02-08T17:25:29.266+00:00][INFO ][status] Kibana is now available (was degraded)
```

**Comment by ph** [comment]
OK, I think we might have two different problems, let's start with the APM-Server, recently we have removed the autogeneration of configuration of fleet-server without a human 'intervention' https://github.com/elastic/kibana/issues/108456.  Looking at the APM docker-compose file at https://github.com/elastic/apm-server/blob/main/docker-compose.yml#L41-L63 we never configure the default fleet server configuration. So this aligns with we what see in the log Fleet Server is waiting on a configuration that will never exist.  Elastic Package has creared a PR with https://github.com/elastic/elastic-package/pull/676

Now I will check with the elastic-package.

**Comment by ph** [comment]
When I've tested https://github.com/elastic/fleet-server/issues/1129#issuecomment-1029344378 I didn't use the **container** subcommand and did use the link from the kibana UI, so in that case Kibana generates the appropriate server configuration.

**Comment by ph** [comment]
Added notes here, using this configuration yield a few deprecation warning from https://github.com/elastic/kibana/issues/108456#issuecomment-1016502179
```
xpack.fleet.agentPolicies:
  - name: Agent policy 1
    description: Agent policy 1
    is_managed: false
    namespace: default
    monitoring_enabled:
      - logs
      - metrics
    package_policies:
      - name: system-1
        id: default-system
        package:
          name: system
  - name: Fleet Server policy preconfigured
    id: fleet-server-policy
    namespace: default
    package_policies:
      - name: Fleet Server
        package:
          name: fleet_server
```

**Comment by ph** [comment]
I still think it's something only when using an automation workflow, when I have a user journey it seems to work at least outside of containers.

**Comment by ph** [comment]
OK, 8.1.0 is stuck in a failure loop on Fleet-Server, the server is not even started. This is exactly what marcin had. 
Going to do the same thing in the 8.2.0 artifacts.

**Comment by ph** [comment]
OK, 8.2.0 elastic-package stack works for me.

```
Creating network "elastic-package-stack_default" with the default driver
Creating elastic-package-stack_elasticsearch_1    ... done
Creating elastic-package-stack_package-registry_1 ... done
Creating elastic-package-stack_package-registry_is_ready_1 ... done
Creating elastic-package-stack_kibana_1                    ... done
Creating elastic-package-stack_elasticsearch_is_ready_1    ... done
Creating elastic-package-stack_fleet-server_1              ... done
Creating elastic-package-stack_kibana_is_ready_1           ... done
Creating elastic-package-stack_elastic-agent_1             ... done
Creating elastic-package-stack_fleet-server_is_ready_1     ... done
Creating elastic-package-stack_elastic-agent_is_ready_1    ... done
Done
```


Logging into Kibana show both Elastic Agent connecting to it. everything seems to be enrolled fine.
I don't know how it's used in the CI but if the 8.2.0 works. I wonder if the asserts or the configuration of the integration has issues in the CI.


@jlind23 @axw The main difference in 8.2.0 and 8.1.0 is really the instrumentation. fleet-server is identical.

**Comment by mtojek** [comment]
Thanks, @ph, for working on this to reduce the blast.

I opened a similar PR to verify the 8.2.0 stack: https://github.com/elastic/elastic-package/pull/692

Hey @simitt @axw @stuartnelson3, I suppose you've been already researching the APM instrumentation issue. Could you please share more details or link the issue, so we can learn what went wrong here? My bet is an undetected library conflict somewhere around GRPC.

**Comment by ph** [comment]
Elastic-package and apm-server problem is fixed so I am going to close this issue, If there is a problem we can reopen it.

**Comment by simitt** [comment]
> Hey @simitt @axw @stuartnelson3, I suppose you've been already researching the APM instrumentation issue. Could you please share more details or link the issue, so we can learn what went wrong here? My bet is an undetected library conflict somewhere around GRPC.

@stuartnelson3 is looking into this.
